Client/Shared/NavMenu.razor

```diff
@@ -0,0 +1,112 @@
+﻿@using AndcultureCode.CSharp.Core.Extensions
+@using BlazorCMS.Client.Services
+@using BlazorCMS.Client.State
+@using BlazorCMS.Shared.Dtos
+@inherits BlazorState.BlazorStateComponent
+
+<div class="top-row pl-4 navbar navbar-dark">
+    <a class="navbar-brand" href="">BlazorCMS</a>
+    <button class="navbar-toggler" @onclick="ToggleNavMenu">
+        <span class="navbar-toggler-icon"></span>
+    </button>
+</div>
+
+<div class="@NavMenuCssClass" @onclick="ToggleNavMenu">
+    <ul class="nav flex-column">
+        <li class="nav-item px-3">
+            <NavLink class="nav-link" href="" Match="NavLinkMatch.All">
+                <span class="oi oi-home" aria-hidden="true"></span> Home
+            </NavLink>
+        </li>
+        @foreach (var section in Sections)

```

Comment:
it looks like you're gonna have some classic n+1 issues here. Is there another way you could compose this markup? 

Likely won't be a big deal, but something worth noting from a performance standpoint!